app/src/modules/insights/components/workspace.vue

```diff
@@ -117,12 +118,24 @@ export default defineComponent({
 			};
 		});
 
-		return { workspaceSize, workspaceBoxSize, mainElement, zoomScale };
+		return { workspaceSize, workspaceBoxSize, mainElement, zoomScale, getSimpleHashForPanel };
 
 		function getVar(cssVar: string) {
 			if (!mainElement.value) return;
 			return getComputedStyle(mainElement.value).getPropertyValue(cssVar).trim();
 		}
+
+		function getSimpleHashForPanel(panel: Panel & { _coordinates: [number, number][] }) {
+			const trackedProperties = omit(panel, ['position_x', 'position_y', '_coordinates']);
+			const str = JSON.stringify(trackedProperties);
+
+			let hash = 0;
+			for (let i = 0; i < str.length; hash &= hash) {
+				hash = 31 * hash + str.charCodeAt(i++);
+			}
+
+			return Math.abs(hash).toString(16);
```

Comment:
@azrikahar There's the `get-simple-hash` utils in API, https://github.com/directus/directus/blob/main/api/src/utils/get-simple-hash.ts
Will it be better to shift the above lib into the `@directus/shared` instead?

Reply:
Yea this was indeed grabbed from there, and I did thought about it but there's also some other functions that have yet to be moved to shared so I didn't end up moving it there. 😄 Hmm I guess there's no better time than to start now. I'll proceed to move it to shared then.